src/main/scala/firrtl/annotations/Annotation.scala

```diff
@@ -77,6 +77,28 @@ trait SingleTargetAnnotation[T <: Named] extends Annotation {
   }
 }
 
+trait MultiTargetAnnotation extends Annotation {
```

Comment:
This needs ScalaDoc describing its behavior.

In addition, what should the desired behavior be if the two targets are renamed to a different number of new targets? I wonder if there is a better way to represent this, perhaps as a Seq[Seq[Target]], where each target is expanded (flatMap) within its container Seq?

E.g. Seq(Seq(a), Seq(b)) -> Seq(Seq(a1, a2), Seq(b1, b2, b3))

Food for thought. I agree though that we should certainly add something like this - I've found myself duplicating this code over and over again....

Reply:
I try to folded `Seq[Seq[Target]]` to `Seq[Target]` is aimed to give a same behavior with `SingleTargetAnnotation`. If you permit `Seq[Seq[Target]]` would be much better.
 I think `crossJoin` is also a useful function for the multi target folding in `RenameMap`, so do mind I keeping them as a private function with detail documentation?